sklearn/tests/test_dummy.py

```diff
@@ -401,27 +404,34 @@ def test_quantile_invalid():
     y = [0] * 5  # ignored
 
     est = DummyRegressor(strategy="quantile")
-    assert_raises(ValueError, est.fit, X, y)
+    with pytest.raises(ValueError):
+        est.fit(X, y)
 
     est = DummyRegressor(strategy="quantile", quantile=None)
-    assert_raises(ValueError, est.fit, X, y)
+    with pytest.raises(ValueError):
+        est.fit(X, y)
 
     est = DummyRegressor(strategy="quantile", quantile=[0])
-    assert_raises(ValueError, est.fit, X, y)
+    with pytest.raises(ValueError):
+        est.fit(X, y)
 
     est = DummyRegressor(strategy="quantile", quantile=-0.1)
-    assert_raises(ValueError, est.fit, X, y)
+    with pytest.raises(ValueError):
+        est.fit(X, y)
 
     est = DummyRegressor(strategy="quantile", quantile=1.1)
-    assert_raises(ValueError, est.fit, X, y)
+    with pytest.raises(ValueError):
+        est.fit(X, y)
 
     est = DummyRegressor(strategy="quantile", quantile='abc')
-    assert_raises(TypeError, est.fit, X, y)
+    with pytest.raises(ValueError):
```

Comment:
```suggestion
    with pytest.raises(TypeError):
```

Reply:
Optionally, the whole test could be parameterizes as:
```python
@pytest.mark.parametrize(
    "quantile, error",
    [
        ("default", ValueError),
        (None, ValueError),
        # TODO: Fill in the futher cases here
    ],
)
def test_quantile_invalid(quantile, error):
    X = [[0]] * 5  # ignored
    y = [0] * 5  # ignored

    if quantile == "default":
        est = DummyRegressor(strategy="quantile")
    else:
        est = DummyRegressor(strategy="quantile", quantile=quantile)
    with pytest.raises(ValueError):
        est.fit(X, y)
```

Reply:
I am not sure parametrization helps with readability in this case.